src/api/app/views/source/verboseproductlist.xml.builder

```diff
@@ -1,9 +1,8 @@
-# avoid to render, parser and re-render here, since it can be a hughe content
-
-output = ''
-output << "<productlist count='#{@products.length}'>\n"
-@products.each do |p|
-  output << p.to_axml
+xml.productlist(count: @products.count) do
+  @products.map do |p|
```

Comment:
Please don't use single chars for variable names unless it's a one liner. Naming it `package` makes the code much more readable.